ci_framework/roles/cifmw_cephadm/tasks/check_vip.yml

```diff
@@ -0,0 +1,33 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Process VIP
+  block:
+    - name: Increment the retry count
+      set_fact:
+        count: "{{ 2 if count is undefined else count | int + 2 }}"
+
+    - name: Get an IP address from the Storage network
+      set_fact:
+        cur_ip: "{{ storage_network_range | ansible.utils.next_nth_usable(count) }}"
+
+    - name: Reserve VIP if the address is available
+      set_fact:
+        cifmw_cephadm_vip: "{{ cur_ip }}"
+      when: cur_ip not in ips | default([])
+  when:
+    # cifmw_cephadm_vip is always defined in the ceph.yml playbook
+    - cifmw_cephadm_vip | default() | length == 0
```

Comment:
@fultonj this condition is important because when I find an available `VIP` I `stop` the loop making `noop` the remaining executions (just because `breaking` the loop like this is not available in ansible).
Anyway the number of iterations here are proportional to the number of `IPs` assigned to the `edpm` nodes.